### ISSUE TITLE
[python3 branch] pyi-grab_version and --version-file not working?

### ISSUE THREAD

**Issue by pandel** [pyinstaller/pyinstaller#1347]
Hi!

It seems as if the version grabbing tool `pyi-grab_version` is not working under python3 branch. Is it possible that setting a decent version info via `--version-file` also isn't working?

I tried with the `test_pkg_structures-version.txt` file from the test suite, but the version info wasn't written to the exe.

Current: 3.0dev-1d241e8

Regards,
Holger

**Comment by KipperK** [comment]
I'm having the same problem here with Python 3.4 and PyInstaller 3.0dev-345ab15 from the download page.  Here is a sample traceback for pyi-grab_version: http://pastebin.com/9q9Vgh9m

That exact command works with the same PyInstaller build if I use Python 2.7.

**Comment by AcidWeb** [comment]
Well reason is quite simple. utils/win32/versioninfo.py is not Python3 ready.

pywintypes.Unicode() don't work correctly with Python3. 

Replace ```pywintypes.Unicode(text)``` with ```text``` (if string is needed) or ```bytes(text, encoding)``` (if bytes is needed).

After that that entire code will probably fail due tons of concat errors but it will be start to making it right.

But making that code work on both Python 2 and 3 will be pain in the ass.

**Comment by SinHouse** [comment]
I solve it using Simple Version Resource Tool 

This is an example:
`verpatch.exe script.exe 1.0.0.0 /va /pv 1.0.0.0 /s description "Your product description" /s product "Your product name" /s copyright "Your company name, 2016" /s company "Your company name"`

[Simple Version Resource Tool](http://www.codeproject.com/Articles/37133/Simple-Version-Resource-Tool-for-Windows)

**Comment by dessant** [comment]
I'm getting these errors with python 3.4 while trying to grab and set the version info, it works as expected with python 2.
```python
$ pyi-grab_version example1.exe
Traceback (most recent call last):
  File "C:\Python34\Scripts\pyi-grab_version-script.py", line 9, in <module>
    load_entry_point('PyInstaller', 'console_scripts', 'pyi-grab_version')()
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\cliutils\grab_version.py", line 36, in run
    vs = PyInstaller.utils.win32.versioninfo.decode(args.exe_file)
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\win32\versioninfo.py", line 112, in decode
    j = vs.fromRaw(data)
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\win32\versioninfo.py", line 136, in fromRaw
    i, (sublen, vallen, wType, nm) = parseCommon(data)
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\win32\versioninfo.py", line 199, in parseCommon
    i, text = parseUString(data, i, i+wLength)
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\win32\versioninfo.py", line 208, in parseUString
    text = unicode(data[start:i], 'UTF-16LE')
NameError: name 'unicode' is not defined

$ pyi-set_version file_version_info.txt example2.exe
Traceback (most recent call last):
  File "C:\Python34\Scripts\pyi-set_version-script.py", line 9, in <module>
    load_entry_point('PyInstaller', 'console_scripts', 'pyi-set_version')()
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\cliutils\set_version.py", line 31, in run
    vs = PyInstaller.utils.win32.versioninfo.SetVersion(exe_file, info_file)
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\win32\versioninfo.py", line 558, in SetVersion
    win32api.UpdateResource(hdst, RESOURCE_TYPE['RT_VERSION'], 1, vs.toRaw())
  File "e:\dev\prs\pyinstaller\PyInstaller\utils\win32\versioninfo.py", line 161, in toRaw
    nm = pywintypes.Unicode(u'VS_VERSION_INFO')
TypeError: must be impossible<bad format char>, not str
```

**Comment by htgoebel** [comment]
Will be solved by #2623.
